app/coffee/RedisManager.coffee

```diff
@@ -70,11 +75,12 @@ module.exports = RedisManager =
 			callback null, len
 
 	setDocument : (doc_id, docLines, version, callback = (error) ->)->
-		multi = rclient.multi()
-		multi.set keys.docLines(doc_id:doc_id), JSON.stringify(docLines)
-		multi.set keys.docVersion(doc_id:doc_id), version
-		multi.incr keys.now("docsets")
-		multi.exec (error, replys) -> callback(error)
+		ZipManager.compressIfNeeded doc_id, JSON.stringify(docLines), (err, result) ->
+			multi = rclient.multi()
+			multi.set keys.docLines(doc_id:doc_id), result
```

Comment:
Correct, it is happy to take a buffer.  It's not explicitly documented but the code has explicit cases for buffer and string arguments as input.

https://github.com/mranney/node_redis/issues/46
"Regardless of how "return_buffers" is set, you can always supply either a Buffer or a String to a command."

Incidentally the author of node_redis suggests using your 2-client solution here:
https://github.com/mranney/node_redis/issues/109